lib/craft/machine.ex

```diff
@@ -440,6 +441,18 @@ defmodule Craft.Machine do
     {:reply, %{state: state, machine_state: machine_state}, state}
   end
 
+  @impl true
+  def handle_info({:user_message, msg}, state) do
+    private =
+      if function_exported?(state.module, :handle_info, 2) do
+        state.module.handle_info(msg, state.private)
+      else
+         raise "Received message but #{state.module} has no `handle_info/2` defined"
```

Comment:
we should just ignore the message, since `handle_info/2` is an optional callback